scrapers/scraper.py

```diff
@@ -0,0 +1,267 @@
+        try:
+            sys.stdout = open(os.devnull, "w")
+            stems = countLexcStems(lexcFile)
+        except SystemExit:
+            continue
+        finally:
+            sys.stdout = sys.__stdout__
+
+        commitData["stems"] = stems
+        history.append(commitData)
+
+    return {"name": language, "history": history}
+
+def pairHistory(language, packages):
+    """Returns the history of all pairs of a language"""
+    langPackages = []
+    for package in packages:
+        if not(language in package["name"] and re.match(r"apertium-\w+-\w+", package["name"])):
+            continue
+
+        pairName = package["name"].replace("apertium-", "")
+        pairList = pairName.split("-")
+        if not set(pairList) <= set(languages):
+            continue
+
+        dixName = pairName if pairName != "tat-kir" else "tt-ky"
+        subprocess.call(['git', 'clone', 'https://github.com/apertium/apertium-{}'.format(pairName)], stderr=subprocess.DEVNULL)
+        shutil.copyfile("./.mailmap", "./apertium-{}/.mailmap".format(pairName))
+        commits = subprocess.check_output(['git', 'log', '--use-mailmap', '--follow', 'apertium-{0}.{0}.dix'.format(dixName)], cwd='./apertium-{}/'.format(pairName)).decode('utf-8').split('\n\n')
+        shutil.rmtree("./apertium-{}/".format(pairName), onerror=del_rw)
+
+        try:
+            oldFile = json.loads(open("../json/{}.json".format(language), "r").read())
+            for data in oldFile:
+                if data["name"] == pairName:
+                    history = data["history"]
+                    break
+                history = []
+        except FileNotFoundError:
+            history = []
+
+        for commit in commits:
+            if commit.startswith(' '):
+                continue
+            commitData = {}
+            lines = commit.split('\n')
+            for line in lines:
+                if line.startswith('commit'):
+                    commitData["sha"] = line[len("commit "):]
+                if line.startswith('Author'):
+                    commitData["author"] = line.split(": ")[1].split(" <")[0]
+                if line.startswith('Date'):
+                    commitData["date"] = line[len("date:"):].strip()
+
+            if any(commitData["sha"] == cm["sha"] for cm in history):
+                continue
+
+            dixFile = "https://raw.githubusercontent.com/apertium/apertium-{0}/{1}/apertium-{0}.{0}.dix".format(pairName, commitData["sha"])
+            stems = countDixStems(dixFile, True)
+            if stems == -1:
+                dixName = iso3to2[pairList[0]] + "-" + iso3to2[pairList[1]]
+                dixFile = "https://raw.githubusercontent.com/apertium/apertium-{0}/{2}/apertium-{1}.{1}.dix".format(pairName, dixName, commitData["sha"])
+                stems = countDixStems(dixFile, True)
+                if stems == -1:
+                    continue
```

Comment:
I'd almost suggest logging something for these exceptional cases. Perhaps a warning. Scripts don't need to be completely silent!